src/OrchardCore.Modules/OrchardCore.Alias/Services/AliasShapeTableProvider.cs

```diff
@@ -29,6 +29,25 @@ public override ValueTask DiscoverAsync(ShapeTableBuilder builder)
                     displaying.Shape.Metadata.Alternates.Add("Content_" + displaying.Shape.Metadata.DisplayType + "__Alias__" + encodedAlias);
                 }
             });
+        builder.Describe("Widget")
+            .OnDisplaying(displaying =>
+            {
+                var shape = displaying.Shape;
+                var contentItem = shape.GetProperty<ContentItem>("ContentItem");
+
+                var aliasPart = contentItem?.As<AliasPart>();
+
+                if (aliasPart != null)
+                {
+                    var encodedAlias = aliasPart.Alias.EncodeAlternateElement();
+
+                    // Widget__Alias__[Alias] e.g. Widget-Alias-example, Widget-Alias-my-page
+                    displaying.Shape.Metadata.Alternates.Add("Widget__Alias__" + encodedAlias);
+
+                    // Widget_[DisplayType]__Alias__[Alias] e.g. Widget-Alias-example.Summary, Widget-Alias-my-page.Summary
+                    displaying.Shape.Metadata.Alternates.Add("Widget_" + displaying.Shape.Metadata.DisplayType + "__Alias__" + encodedAlias);
+                }
+            });
```

Comment:
Apart from the alternate names, this is the same code as above for `Content`. DRY it instead to a method. Same for `AutorouteShapeTableProvider`.

Reply:
done